test/lib/usd/translators/testUsdExportFileFormat.py

```diff
@@ -0,0 +1,143 @@
+#!/pxrpythonsubst
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import unittest
+
+from maya import cmds
+from maya import standalone
+
+from pxr import Usd
+from pxr import Sdf
+
+import fixturesUtils
+
+class testUsdExportFileFormat(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        fixturesUtils.setUpClass(__file__)
+        cmds.polySphere()
+
+    @classmethod
+    def tearDownClass(cls):
+        standalone.uninitialize()
+
+    def GetUnderlyingUsdFileFormat(self, layer):
+        if Sdf.FileFormat.FindById('usda').CanRead(layer.identifier):
+            return 'usda'
+        elif Sdf.FileFormat.FindById('usdc').CanRead(layer.identifier):
+            return 'usdc'
+        return ''

```

Comment:
Took me some time to figure out how to check what the .usd file format is from Python, the c++ class that also does this doesn't look like it's exposed in Python, but I did find this in a similar core USD test.

Reply:
@fowlertADSK  Nice find!